### DISCUSSION TITLE
How to configure for a testing suite where data needs to be inserted before run

### DISCUSSION THREAD

**Discussion by semoal** [questdb/questdb#5011] [Ideas]
Hi! 

We're migrating from postgres to questdb in our timeseries tables, but we're struggling to configure the tables to **insert the data immediately**. 

We know that Questdb does asynchronous inserting/batching etc..  that's amazing for our production usage but for testing we need the data to be inserted immediately or the promise of the insert wait until it's completely inserted, since we test that with these prices/values inserted our calculations are ok.

I've tried to alter the `o3MaxLag`, `uncommitedRows` or changing the set type wall to bypass, but none worked.

**Comment by ideoma** [comment]
For testing purposes you can use

```SQL
select wait_wal_table('x');
```

This is very crude way of waiting, please don't use it for production load
